cli_tools/gce_onestep_image_import/onestep_importer/aws_args.go

```diff
@@ -0,0 +1,169 @@
+	exportFolder   string
+	exportKey      string
+	exportFileSize int64
+}
+
+// Flags
+const (
+	awsImageIDFlag           = "aws_image_id"
+	awsExportLocationFlag    = "aws_export_location"
+	awsAccessKeyIDFlag       = "aws_access_key_id"
+	awsSecretAccessKeyFlag   = "aws_secret_access_key"
+	awsSessionTokenFlag      = "aws_session_token"
+	awsRegionFlag            = "aws_region"
+	awsExportedAMIPathFlag   = "aws_exported_ami_path"
+	awsResumeExportedAMIFlag = "resume_exported_ami"
+)
+
+var (
+	bucketNameRegex = `[a-z0-9][-_.a-z0-9]*`
+	s3PathRegex     = regexp.MustCompile(fmt.Sprintf(`^s3://(%s)(\/.*)?$`, bucketNameRegex))
+)
+
+// newAWSImportArguments creates a new AWSImportArgument instance.
+func newAWSImportArguments(args *OneStepImportArguments) *awsImportArguments {
+	return &awsImportArguments{
+		accessKeyID:        args.AWSAccessKeyID,
+		executablePath:     args.ExecutablePath,
+		exportLocation:     args.AWSExportLocation,
+		exportedAMIPath:    args.AWSExportedAMIPath,
+		gcsComputeEndpoint: args.ComputeEndpoint,
+		gcsProject:         args.Project,
+		gcsZone:            args.Zone,
+		gcsRegion:          args.Region,
+		gcsScratchBucket:   args.ScratchBucketGcsPath,
+		gcsStorageLocation: args.StorageLocation,
+		imageID:            args.AWSImageID,
+		region:             args.AWSRegion,
+		resumeExportedAMI:  args.AWSResumeExportedAMI,
+		secretAccessKey:    args.AWSSecretAccessKey,
+		sessionToken:       args.AWSSessionToken,
+	}
+}
+
+// ValidateAndPopulate validates args related to import from AWS, and populates
+// any missing parameters.
+func (args *awsImportArguments) validateAndPopulate(populator param.Populator) error {
+	err := args.validate()
+	if err != nil {
+		return err
+	}
+
+	err = populator.PopulateMissingParameters(&args.gcsProject, &args.gcsZone, &args.gcsRegion,
+		&args.gcsScratchBucket, "", &args.gcsStorageLocation)
+	if err != nil {
+		return err
+	}
+
+	return args.getS3PathElements()
+}
+
+func (args *awsImportArguments) validate() error {
+	if err := validation.ValidateStringFlagNotEmpty(args.accessKeyID, awsAccessKeyIDFlag); err != nil {
+		return err
+	}
+	if err := validation.ValidateStringFlagNotEmpty(args.secretAccessKey, awsSecretAccessKeyFlag); err != nil {
+		return err
+	}
+	if err := validation.ValidateStringFlagNotEmpty(args.region, awsRegionFlag); err != nil {
+		return err
+	}
+
+	if args.resumeExportedAMI {
+		if args.exportedAMIPath == "" {
+			return fmt.Errorf("To resume exported AMI, flag -%v must be provided", awsExportedAMIPathFlag)
+		}
+	} else {
+		if args.imageID == "" || args.exportLocation == "" {
+			return fmt.Errorf("To export AMI, flags -%v and -%v must be provided", awsImageIDFlag, awsExportLocationFlag)
+		}
+	}
+
+	return nil
+}
+
+// getS3PathElements gets bucket name, and folder or object key depending on if
+// AMI has been exported, for a valid object path. Error is returned otherwise.
+func (args *awsImportArguments) getS3PathElements() error {
```

Comment:
generaste is better than get in my opinion

Reply:
yep, I agree, I'll make the change.